### ISSUE TITLE
Unify comments

### ISSUE THREAD

**Issue by tsujigiri** [gedankenstuecke/snpr#143]
Currently there are three models for commenting different other models. In order to decrease complexity and duplication, let's unify these to one `Comment` model with a polymorphic association to `commentable`. Let's also clean up the views.

**Write tests!**

**Comment by tsujigiri** [comment]
I started working on this: https://github.com/tsujigiri/snpr/tree/one_model_to_comment_them_all

**Comment by philippbayer** [comment]
Just stumbled over this nice blog-entry: http://ivory.idyll.org/blog/2015-using-hypothesis-on-workshop-sites.html

tl;dr: bind in a piece of javascript from hypothes.is, you get a new menu on the side, users can highlight sections and comment on them specifically

That would enable us to

a) have unified comments (comment *all* the things)
b) have comments on very specific parts of the site - for example, a user found out that the genotypes listed for rs1234 are flipped and should actually be A/T, or a position is wrong, or my avatar looks stupid, or ....

Also, annotations common to pieces of text are shown wherever the piece of text is shown (not sure whether we have this case)

**Comment by tsujigiri** [comment]
While this is nice and generic thing, I'm not sure about the not being in control of the comments as well as having the users have to register another account in order to be able to comment. It's a pretty big step in terms of usability.

**Comment by gedankenstuecke** [comment]
I agree with @tsujigiri on this one. Having to register for another account in order to be able to write comments seems too much effort for most people I'd argue. Especially as the comments are already not used too much.

**Comment by gedankenstuecke** [comment]
Maybe we could tackle this problem soonish. It seems the comments are broken right now. I tried writing a SNP comment and it crashed with a 500 error in production :wink:
